### DISCUSSION TITLE
Set the general settings (temperature, top_k, etc.) at prompt or chat level

### DISCUSSION THREAD

**Discussion by denispol** [open-webui/open-webui#1669] [Ideas]
Hi, I was thinking, since the settings we use (like temperature, top_k, top_p, etc.) really depend on what we're trying to do, it would be  helpful to be able to tweak them on a per-conversation basis. In a similar fashion as how Lobe-chat lets you adjust the temperature for each new chat. For example, when I'm coding, I need the temperature to be pretty low so I don't get many mistakes, but that's not ideal for other use cases where I want the AI to be more creative.

Implementation of:

- Temperature and/or other settings at the chat level, using a button with a slider
- Default model, temperature and other settings at the prompt level

**Comment by dagelf** [comment]
Seconded. Actually, on a per response basis - add to that, the ability to generate 2 or more responses to the same query, in paralleled, the way Gemini does it, so that you can choose among them.

**Comment by masc-it** [comment]
Newcomer question: how do you even set these settings in general? The docs state the fact we can teak them, but... how? I cannot find any GUI section that let me tweak temperature, sampling, etc.. am I blind?

**Reply by delfireinoso** [reply]
On **Settings, General, Advanced Parameters**

**Comment by LeroyDaydreamer** [comment]
These settings including the system prompt do not belong into a global settings area at all! They are highly dependend on the current LLM and the task you are performing. Technically they can indeed even be changed per prompt. So not having them available in chat or at least when selecting the LLM is quite annoying and confusing. If the profile contains these settings on user level they should only serve as default. So whenever a new LLM is used for the first time, a new profile for the LLM should be created containing the system prompt for that specific LLM and so on, so the values from the user profile are only used once for initialization of a new LLM. The LLM settings should be available at all times. Indeed the settings could again be overridden by chat or prompt specific settings. Implementing such a hierarchy makes it a lot more complicated of course so taking the settings out of the user settings and moving them to LLM specific parameters would already be a major improvement. If I understand it correctly though there is are currently no way to associate meta data with a specific LLM. So yes, it may indeed be more complicated than it sounds to implement LLM specific settings. Still having to use the same settings for different LLM doesn't really work and doesn't make sense either. 

The only way to set LLM specific settings now that I'm aware of is to create new model files. Why Ollama doesn't differentiate between models (weights + config) and common LLM parameters (prompt, temperature etc.) is beyond me. Possibly open-webui could do it in a transparent way, like creating a new model file with a suffix like _webui and just not displaying it in the list of models. So when model XYZ is selected, actually "model" XYZ_webui will be loaded and if it doesn't exist yet, it will be created. If then the prompt is changed for example, the modelfile gets updated. Of course this approach only works with Ollama as backend and is not very elegant either.
